### PR TITLE
made changes to data types and responses

### PR DESCRIPTION
**Pull request by prayujt**
- Added `difficulty`, `author`, and `codeSolution` fields to problems.
- Changed getProblem to not return test case info in student call
- Moved language specification to courses instead of problems

### CODE REVIEW

File: src/api/dal/problem.ts [hunk 1]
```diff
@@ -81,11 +90,19 @@ export const getByModule = async (
...
+  const args = {
+    where: {
+      moduleId: moduleId
+    }
+  };
+
+  if (!hidden) {
+    args['where']['hidden'] = hidden;
+    args['include'] = 'testCases';
+    args['order'] = [['testCases', 'orderNumber', 'ASC']];
```
Comment (kapooramanpreet) [inline src/api/dal/problem.ts:102]: do we also include solution here?

**Review by kapooramanpreet — CHANGES_REQUESTED** [review]

File: src/api/models/v2/problem.model.ts [hunk 2]
```diff
@@ -12,11 +12,13 @@ export interface ProblemAttributes {
...
   codeBody: string;
   codeFooter: string;
+  codeSolution: string;
   templatePackage: string;
   timeLimit: number;
   memoryLimit: number;
   buildCommand: string;
-  languages: string;
+  difficulty: string;
+  author: string;
```
[the remark this answers is not in the document]
Comment (prayujt) [reply src/api/models/v2/problem.model.ts:21]: planetscale automatically includes the `dateCreated` and `dateModified` properties - assuming we stick with planetscale we don't need to handle that on our end and it is returned when retrieving the problem/lesson

**Review by digitdustin — DISMISSED** [review]

**Review by kapooramanpreet — DISMISSED** [review]
LG!

**Review by kapooramanpreet — APPROVED** [review]
LG!

**Review by marcgabe15 — APPROVED** [review]

**Review by kapooramanpreet — COMMENTED** [review]
LG

**Pull request merged**